### ISSUE TITLE
v2.3-dev fails to compile on OSX with clang

### ISSUE THREAD

**Issue by sdwfrost** [veg/hyphy#497]
Compiling v2.3-dev on OSX gives the following error:

```bash
hyphy/src/core/calcnode.cpp:6241:67: error: expected expression
      site_has_all_gaps &= (!ArrayAny (iterator->theProbs, cBase, [](_Pa...
                                                                  ^
```

The output of `clang -v`

```bash
Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin15.3.0
Thread model: posix
```

**Comment by sdwfrost** [comment]
I also tried updating to XCode 8, but I get the same error. Compiler details below.

```bash
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
```

**Comment by spond** [comment]
Dear @sdwfrost,

Thank you for reporting the issue. Resolved via caf745b

Best,
Sergei
